lib/datadog/tracing/contrib/patcher.rb

```diff
@@ -55,6 +55,14 @@ def on_patch_error(e)
             Datadog.health_metrics.error_instrumentation_patch(1, tags: tags)
           end
 
+          def patch_results
+            @patch_results ||= nil
+          end
+
+          def patch_results=(patch_results)
+            @patch_results = patch_results
```

Comment:
Another issue I found is that `patch_results` is just a boolean today, when I tested this in my terminal.
We need to collect the error information similar to here https://github.com/DataDog/dd-trace-rb/blob/54c5e1ee69df3089bc9f73c4c6f01d5046b4f004/lib/datadog/tracing/contrib/patcher.rb#L49 and here https://github.com/DataDog/dd-trace-rb/blob/54c5e1ee69df3089bc9f73c4c6f01d5046b4f004/lib/datadog/tracing/contrib/patcher.rb#L53
And store it for consumption.

I don't think we need to store anything for a successful patch, thus I suggest we rename `patch_results` to `patch_error_result`.
I recommend returning:

```ruby
            {
              type: e.class.name,
              message: e.message,
              line: Array(e.backtrace).first
            }
```

Reply:
We should test a case where the Patcher failed to run, to ensure we are capturing exceptions properly.